src/commands/setup/apply-gradle-plugin/inject-plugin-in-build-gradle.ts

```diff
@@ -47,7 +89,7 @@ export const injectPluginInBuildGradle = async (
   if (!hasAddedPluginAndConfiguration) {
     throw new GradlePluginNotInstalled();
   }
-  if (!hasAddedAutomation) {
+  if ((hasAddedAutomationState as AutomationState) !== AutomationState.DONE) {
```

Comment:
is cast needed here? we already know the type `hasAddedAutomationState` (at the declaration site)

Reply:
I agree with you, but Typescript disagrees 😅 

I believe since we only conditionally change the value of `hasAddedAutomationState`, out of the `if` conditions Typescript assumes that its type is `AutomationState.BEFORE_ANDROID_CLOSURE`. 

I did not do a lot of research into that and fixed it quickly, I will ask some typescript teams :)